modules/core/src/main/scala/org/scalasteward/core/nurture/NurtureAlg.scala

```diff
@@ -113,6 +116,40 @@ final class NurtureAlg[F[_]](implicit
       }
     } yield result
 
+  def closeObsoletePullRequests(data: UpdateData): F[Unit] = {
+    val crossDependency = data.update match {
+      case Update.Single(crossDependency, _, _) =>
+        crossDependency
+      case Update.Group(crossDependencies, _) =>
+        crossDependencies.head
+    }
```

Comment:
I would move that inside the implementation of `getObsoleteOpenPullRequests`. That method should just take a `Repo` and an `Update` and return all obsolete PRs.

Reply:
will do